### PR TITLE
Add an additional _defaults_ chart to deploy specific instances of custom resources

### LINKED ISSUE #438: elemental-operator installation sometimes fails
In the CI, we can see that the elemental-operator (dev) installation fails sporadically with this message:

```
msg: "[helm upgrade --install elemental-operator oci://registry.opensuse.org/isv/rancher/elemental/dev/charts/rancher/elemental-operator-chart --namespace cattle-elemental-system --create-namespace] cmd, failed with the following error: Release \"elemental-operator\" does not exist.Installing it now.\nPulled: registry.opensuse.org/isv/rancher/elemental/dev/charts/rancher/elemental-operator-chart:1.2.2\nDigest: sha256:66f55a119c4ac7e3d5aed309b8510fd3288d4a027fef934b37a8e5b20b29f80e\nError: failed post-install: unable to build kubernetes object for deleting hook elemental-operator/templates/channel-hook.yaml: resource mapping not found for name: \"elemental-teal-channel\" namespace: \"fleet-default\" from \"\": no matches for kind \"ManagedOSVersionChannel\" in version \"elemental.cattle.io/v1beta1\"\nensure CRDs are installed first\n",
```

Failed job:
https://github.com/rancher/elemental/actions/runs/4887884892/jobs/8725003338

It also fails when the operator is uninstalled as well:
https://github.com/rancher/elemental/actions/runs/4889180147/jobs/8727492483

### PR DESCRIPTION
**Pull request by davidcassany**
Fixes #438

### CODE REVIEW

File: .obs/chartfile/defaults/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+icon: https://raw.githubusercontent.com/rancher/elemental/main/logo/icon-elemental.svg
+version: "%VERSION%"
+appVersion: "%VERSION%"
+annotations:
+  catalog.cattle.io/certified: rancher
+  catalog.cattle.io/display-name: Elemental Operator default resources
+  catalog.cattle.io/experimental: "true"
+  catalog.cattle.io/namespace: cattle-elemental-system
+  catalog.cattle.io/os: linux
+  catalog.cattle.io/provides-gvr: elemental.cattle.io/v1beta1
```
Comment (fgiudici) [inline .obs/chartfile/defaults/Chart.yaml:17]: Wondering if we should drop this annotation, as the Elemental GVR is provided by the other chart 🤔 
I was not able anyway to find a doc with Rancher annotations for charts explained.
So, not sure of the implications how having or removing it 🤷🏼 
```suggestion

```

File: charts/defaults/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+icon: https://raw.githubusercontent.com/rancher/elemental/main/logo/icon-elemental.svg
+version: 0.0.0
+appVersion: 0.0.0
+annotations:
+  catalog.cattle.io/certified: rancher
+  catalog.cattle.io/display-name: Elemental Operator default resources
+  catalog.cattle.io/experimental: "true"
+  catalog.cattle.io/namespace: cattle-elemental-system
+  catalog.cattle.io/os: linux
+  catalog.cattle.io/provides-gvr: elemental.cattle.io/v1beta1
```
Comment (fgiudici) [inline charts/defaults/Chart.yaml:13]:
```suggestion

```

File: .obs/chartfile/defaults/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+#!BuildTag: rancher/elemental-operator-defaults-chart:latest
+#!BuildTag: rancher/elemental-operator-defaults-chart:%VERSION%
+#!BuildTag: rancher/elemental-operator-defaults-chart:%VERSION%-%RELEASE%
+apiVersion: v2
+name: elemental-operator-defaults
+description: A Helm chart for deploying Rancher Elemental Operator CRDs
+icon: https://raw.githubusercontent.com/rancher/elemental/main/logo/icon-elemental.svg
```
Comment (fgiudici) [inline .obs/chartfile/defaults/Chart.yaml:8]: minor: I would drop the icon from the defaults chart
```suggestion

```

File: charts/defaults/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v2
+name: elemental-operator-defaults
+description: A Helm chart for deploying Rancher Elemental Operator default resources
+icon: https://raw.githubusercontent.com/rancher/elemental/main/logo/icon-elemental.svg
```
Comment (fgiudici) [inline charts/defaults/Chart.yaml:4]:
```suggestion

```

**Review by fgiudici — APPROVED** [review]
LGTM! just added a couple of comments on the annotations (feel free to ignore).

**Comment by davidcassany** [comment]
Blocked as this has impact on the UI and should be properly discussed. What happens if the UI starts without a default channel?

**Comment by davidcassany** [comment]
Closing this PR in favor of the former approach of splitting the chart with crds and all the rest.

**Pull request closed without merging**